libs/design-system/src/pagination/usePaginationState.ts

```diff
@@ -1,32 +1,38 @@
 import { useEffect, useState } from 'react';
 import { useSearchParams } from 'react-router-dom';
-import { DEFAULT_PAGINATION_PAGE_SIZES, FIRST_PAGE_NUMBER } from './Pagination.const';
+import { DEFAULT_PAGE_SIZE, DEFAULT_PAGINATION_PAGE_SIZES, FIRST_PAGE_NUMBER } from './Pagination.const';
 
 const URL_PARAM_NAME_PAGE_SIZE = 'size';
 const URL_PARAM_NAME_PAGE_NUMBER = 'page';
 export interface IUsePaginationStateOptions {
+  areSearchParamsEnabled?: boolean;
   startingPageNumber?: number;
+  startingPageSize?: number;
   pageSizes?: number[];
   pageSizeParamName?: string;
   pageNumberParamName?: string;
 }
 
 export const usePaginationState = ({
+  areSearchParamsEnabled = true,
   startingPageNumber = FIRST_PAGE_NUMBER,
+  startingPageSize = DEFAULT_PAGE_SIZE,
   pageSizes = DEFAULT_PAGINATION_PAGE_SIZES,
   pageNumberParamName = URL_PARAM_NAME_PAGE_NUMBER,
   pageSizeParamName = URL_PARAM_NAME_PAGE_SIZE,
 }: IUsePaginationStateOptions) => {
   const [searchParams, setSearchParams] = useSearchParams();
 
   const [pageSize, setPageSize] = useState<number>(
-    getValidatedPageSizeFromUrl(searchParams.get(pageSizeParamName), pageSizes)
+    getValidatedPageSizeFromUrl(searchParams.get(pageSizeParamName), startingPageSize, pageSizes)
   );
   const [currentPageNumber, setCurrentPageNumber] = useState<number>(
     getValidatedPageNumberFromUrl(searchParams.get(pageNumberParamName), startingPageNumber)
   );
 
   useEffect(() => {
+    if (!areSearchParamsEnabled) return;
```

Comment:
```suggestion
    if (!areSearchParamsEnabled) { 
       return;
    }
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** Personal preference here (but strong one) -- I don't like inline returns since they can accidentally lead to frustrating behavior when other devs touch the code. I like to always use `{ }` explicitly with `if` blocks to ensure things are scoped properly within